Fix auth connecting to pulp api.

alikins opened the pull request:
galaxy_pup.api_client.APIClient.request()
doesn't set auth headers the same way
other higher level calls do. So make sure
it sets the auth headers.

Otherwise the basic auth connection to pulp_ansible fails on publish

cutwater commented on galaxy_api/api/v3/viewsets.py:
```diff
@@ -124,15 +124,22 @@ def post(self, request, *args, **kwargs):
             host=api.configuration.host,
-            prefix=settings.API_PATH_PREFIX,
-            path='/v3/artifacts/collections/',
+            # prefix='',
```
Please don't leave commented code.

alikins commented:
Fixed up commented code. 

Also split the commit into two commits. One for the URL update, and another for 
setting correct auth headers for the POST request to pulp-api.

(Ideally, the use of the lower level APIClient.request() wouldn't be needed, but I don't
see an obvious way to do that without tweaking the templates used to generate
the galaxy_python).

romanblanco commented:
https://github.com/ansible/galaxy-dev/issues/163
